pysmt/formula.py

```diff
@@ -86,6 +86,11 @@ def _do_type_check(self, formula):
         self._do_type_check = self._do_type_check_real
         return self._do_type_check(formula)
 
+    def _sort_nodes(self, nodes):
+        if len(nodes) == 1 and isinstance(next(iter(nodes)), Iterable):
+            nodes = nodes[0]
+        return sorted(nodes, key=lambda p: p.node_id())
```

Comment:
Could this method return directly a tuple?
It seems that the main way we use it is `tuple(_sort_nodes(...))`.